tests/benchmarks/data_structure_perf/dlist_perf/src/dlist_perf.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief doubly-linked list tests
+ *
+ * @defgroup lib_dlist_tests Dlist
+ */
+
+#include <ztest.h>
+#include <sys/dlist.h>
+
+#define NODE_SIZE 5
+
+static sys_dlist_t test_list;
+
+struct container_node {
+	sys_dnode_t node;
+	int value;
+};
+
+/**
+ * @brief Test whether dlist node struct is embeddedable
+ * in any user struct
+ *
+ * @details
+ * Initialize a user defined struct contains dlist node
+ * Add into a doubly-linked list
+ * Enumerate the doubly-linked list.
```

Comment:
Could you give it more descriptions about the test method? thanks!

Reply:
Hi @enjiamai , I have done this, thanks for your comments. :)